app/models/user.rb

```diff
@@ -4,6 +4,17 @@ class User < ApplicationRecord
 
   has_many :articles, foreign_key: 'author_id', dependent: :destroy
   has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+  has_many :liked_articles, through: :likes, source: :article
 
   validates :username, presence: true, length: { minimum: 2, maximum: 90 }
+
+  def like(article)
+    like = likes.build(article: article)
+    likes << like
+  end
```

Comment:
Can you place this code outside of model? Lets use model only for database mapping, you can create builder class or use interactor to build 'like' object